pkg/virt-handler/options_test.go

```diff
@@ -0,0 +1,176 @@
+				Expect(expectedTopology.NumaCells[i].Distances).To(ConsistOf(matchDistances(cell.Distances.Siblings)))
+				Expect(expectedTopology.NumaCells[i].Cpus).To(ConsistOf(matchCPUs(cell.CPUS.CPUs)))
+			}
+		},
+			Entry("Single cell, single page, single CPU per cell", 1, 1, 1),
+			Entry("Multiple cells, multiple pages, multiple CPUs per cell", 4, 3, 4),
+		)
+	})
+})
+
+type option func(caps *libvirtxml.Caps)
+
+func newLibvirtCapsTopology(opts ...option) *libvirtxml.Caps {
+	caps := &libvirtxml.Caps{
+		Host: libvirtxml.CapsHost{
+			NUMA: &libvirtxml.CapsHostNUMATopology{},
+		},
+	}
+
+	for _, f := range opts {
+		f(caps)
+	}
+
+	return caps
+}
+
+func withCells(count int) option {
+	return func(caps *libvirtxml.Caps) {
+		caps.Host.NUMA.Cells = &libvirtxml.CapsHostNUMACells{
+			Cells: make([]libvirtxml.CapsHostNUMACell, count),
+		}
+		for i := 0; i < count; i++ {
+			caps.Host.NUMA.Cells.Cells[i] = libvirtxml.CapsHostNUMACell{
+				ID: i,
+			}
+		}
+	}
+}
+
+func withCellMemory(size uint64, unit string) option {
+	return func(caps *libvirtxml.Caps) {
+		for i := range caps.Host.NUMA.Cells.Cells {
+			caps.Host.NUMA.Cells.Cells[i].Memory = &libvirtxml.CapsHostNUMAMemory{
+				Unit: unit,
+				Size: size,
+			}
+		}
+	}
+}
+
+func withCellPages(pageCount int, count uint64, unit string) option {
+	return func(caps *libvirtxml.Caps) {
+		for i := range caps.Host.NUMA.Cells.Cells {
+			caps.Host.NUMA.Cells.Cells[i].PageInfo = make([]libvirtxml.CapsHostNUMAPageInfo, pageCount)
+			for j := range pageCount {
+				caps.Host.NUMA.Cells.Cells[i].PageInfo[j] = libvirtxml.CapsHostNUMAPageInfo{
+					Unit:  unit,
+					Count: count,
+					// Similar growth to common hugepages (4KiB, 2MiB. 1GiB)
+					Size: 4 << (9 * j),
+				}
+			}
+		}
+	}
+}
+
+func withCellDistances(distance int) option {
+	return func(caps *libvirtxml.Caps) {
+		for i := range caps.Host.NUMA.Cells.Cells {
+			caps.Host.NUMA.Cells.Cells[i].Distances = &libvirtxml.CapsHostNUMADistances{
+				Siblings: make([]libvirtxml.CapsHostNUMASibling, len(caps.Host.NUMA.Cells.Cells)),
+			}
+			for j := range caps.Host.NUMA.Cells.Cells {
+				// Doesn't mimic standard NUMA behavior, no differing latency
+				caps.Host.NUMA.Cells.Cells[i].Distances.Siblings[j] = libvirtxml.CapsHostNUMASibling{
+					ID:    j,
+					Value: distance,
+				}
+			}
+		}
+	}
+}
+
+func withCellCpus(count int) option {
+	return func(caps *libvirtxml.Caps) {
+		for i := range caps.Host.NUMA.Cells.Cells {
+			caps.Host.NUMA.Cells.Cells[i].CPUS = &libvirtxml.CapsHostNUMACPUs{
+				CPUs: make([]libvirtxml.CapsHostNUMACPU, count),
+			}
+			for j := 0; j < count; j++ {
+				// Siblings alternate (e.g. {0,3} {1,2} {2,1} {3,0})
+				currCpu := j + (i * count)
+				caps.Host.NUMA.Cells.Cells[i].CPUS.CPUs[j] = libvirtxml.CapsHostNUMACPU{
+					ID:       currCpu,
+					Siblings: fmt.Sprintf("%d,%d", currCpu, count*(i+1)-j-1),
+				}
+			}
+
+		}
+	}
+}
```

Comment:
Personally, I find hardcoded values easier to follow. The canonical loops in the tests seem a bit complex to me, and I feel we might be overcomplicating things.

Reply:
Done.